internal/pkg/cli/task_run.go

```diff
@@ -302,6 +308,39 @@ func (o *runTaskOpts) Validate() error {
 		}
 	}
 
+	if noOS, noArch := o.os == "", o.arch == ""; noOS != noArch {
+		return fmt.Errorf("must specify either both `--%s` and `--%s` or neither", osFlag, archFlag)
+	}
+	if err := o.validateOS(); err != nil {
+		return err
+	}
+	if err := o.validateArch(); err != nil {
+		return err
+	}
+
+	// Set default values for CPU and memory based on the operating system.
+	// The flag default value is 0. If a user sets `--cpu 0` or `--memory 0`, then the 0 value will be replaced by 256/1024 for cpu and 512/2048 for mem, depending on OS, instead of erroring out.
+	if o.cpu == 0 {
+		o.cpu = 256
+		if isWindowsOS(o.os) {
+			o.cpu = manifest.WindowsTaskCPU
+		}
+	}
+	if o.memory == 0 {
+		o.memory = 512
+		if isWindowsOS(o.os) {
+			o.memory = manifest.WindowsTaskMemory
+		}
+	}
+
+	if o.cpu < 0 {
```

Comment:
We could...but to convert it to a string in the `stack` package we'd have to do:
```
{
			ParameterKey:   aws.String(taskCPUParamKey),
			ParameterValue: aws.String(strconv.FormatInt(int64(t.CPU), 10)),
		}
```
Is that worth avoiding the positive int validation?
Another option is to make sure that it's greater than 256. Is there any case in which it could be 0<cpu<256? Any idea why the validation was just for >0 before?

Reply:
i think in future it is likely they'll allow 0<cpu<256. Maybe just keep as is...